### ISSUE TITLE
Training example

### ISSUE THREAD

**Issue by igorkolesnikov13** [morpheus-project/morpheus#29]
Hi @ryanhausen ,

I'm interested in retraining Morpheus on my data (maybe using some other model) in order to improve results.
I looked into the Morpheus code and saw that there are some functions for training the mode.

Can you provide some jupyter notebook that can walk me through the process of training and substituting the model that Morpheus using.

**Comment by ryanhausen** [comment]
Hi @igorkolesnikov13, thanks taking the time to look into the code! 

The Morpheus model was developed in Tensorflow 1, so I would recommend reimplementing it in Tensorflow 2 if you want to apply the architecture to a new dataset.

Let me know if you have any other questions!

Thanks,

Ryan

**Comment by igorkolesnikov13** [comment]
Yes, this could be a good idea as the next step.

But for now, I wanted to do something simpler. For example, Morpheus is trained on HST data, but I want to use it with SDSS or JWST. So I would like just to retrain the current model.

That's why I wanted to see if there is some plug-in functionality for retraining/replacing the model used for segmentation/morphology.

**Comment by ryanhausen** [comment]
I see, that makes sense. The code currently doesn't support retraining the model on a new dataset in a plug-and-play fashion.

I have reimplemented Morpheus with some improvements in Tensorflow 2, for example the input/output size in 512x512 rather than the 40x40 in this version of Morpheus. Additionally, we have had success applying the newer version of Morpheus to JWST ([ApJL](https://iopscience.iop.org/article/10.3847/2041-8213/aca086)/[arXiv](https://arxiv.org/abs/2208.11456)). I don't think we have tried it on SDSS yet, but I think the difference in pixel scale might affect the results.

I don't plan on adding the ability to retrain this model because it's in Tensorflow1. However, I'll clean up the newer model code and make it public here and I think that you should be able to train that another dataset.

**Comment by mjw1997** [comment]
I read your paper and it is a good idea to apply semantic segmentation to astronomical image analysis. I wanted to reproduce your code, but saw that the code you uploaded didn't have a training process, but a direct demonstration process. I see in your reply above that you will clean up a newer model code and expose it here. I would appreciate it if you could provide us with a jupyter notebook for the training process, or a clean up of the newer model training code.

**Comment by vedantgadekar** [comment]
Hi @ryanhausen,
I wanted to use Morpheus for performing Morphological analysis on JWST images. As you mentioned in your previous posts Morpheus maybe useful on JWST images after retraining. It will be super useful if you could provide us with some insights on how Morpheus can be retrained over a new dataset.

Thanks,
Vedant

**Comment by ryanhausen** [comment]
@vedantgadekar @mjw1997 I apologize for the delayed reply. 

Cleaning up and releasing the newer version of Morpheus is still in progress. I anticipate I will have time in July to get it out. Thanks for your patience.

In the mean time, if you are interested, you can download the publicly available HST fits images and pair them with the publicly available labels from the Morpheus Paper (https://morpheus-project.github.io/morpheus/, go to Data Release) . Then you can build a training and test set from those larger images. Then train the semantic segmentation architecture of your choosing on those samples.

**Comment by Ayush220422** [comment]
Hi @ryanhausen Is the newer version of morpheus where it is trained on the single band image available?
Keen to use the version on JWST images.
